src/providers.ts

```diff
@@ -36,43 +37,38 @@ let unsaved: HTMLParagraphElement;
 const signIn = async (connector: Providers) => {
     let provider: ethers.providers.Web3Provider;
 
-    /**
-     * Connects to the wallet and starts a etherjs provider.
-     */
-    if (connector === 'metamask') {
-        await metamask.request({
-            method: 'eth_requestAccounts',
-        });
-        provider = new ethers.providers.Web3Provider(metamask);
-    } else {
-        /**
-         * The Infura ID provided just for the sake of the demo, you'll need to replace
-         * it if you want to go to production.
-         */
-        walletconnect = new WalletConnect({
-            infuraId: '8fcacee838e04f31b6ec145eb98879c8',
-        });
-        walletconnect.enable();
-        provider = new ethers.providers.Web3Provider(walletconnect);
-    }
-
-    const [address] = await provider.listAccounts();
-    if (!address) {
-        throw new Error('Address not found.');
+    switch (connector) {
+        case Providers.METAMASK:
+            provider = new ethers.providers.Web3Provider(metamask);
+            break;
+        case Providers.WALLET_CONNECT:
+            walletconnect = new WalletConnect({
+                infuraId: "8fcacee838e04f31b6ec145eb98879c8" || process.env.INFURA_API_KEY,
+            });
+            walletconnect.enable();
+            provider = new ethers.providers.Web3Provider(walletconnect);
+            break;
+        default:
+            throw new Error('Unknown connector');
     }
 
-    /**
-     * Try to resolve address ENS and updates the title accordingly.
-     */
-    let ens: string;
+    ssx = new SSX({
+        providers: {
+            web3: { driver: provider },
```

Comment:
We add the instantiated web3 provider here so that SSX can use it for connecting and signing